lib/spack/spack/environment.py

```diff
@@ -175,9 +175,17 @@ def activate(
     # MANPATH, PYTHONPATH, etc. All variables that end in PATH (case-sensitive)
     # become PATH variables.
     #
-    if add_view and default_view_name in env.views:
-        with spack.store.db.read_transaction():
-            cmds += env.add_default_view_to_shell(shell)
+    try:
+        if add_view and default_view_name in env.views:
+            with spack.store.db.read_transaction():
+                cmds += env.add_default_view_to_shell(shell)
+    except (spack.repo.UnknownPackageError,
+            spack.repo.UnknownNamespaceError) as e:
+        tty.error(e)
+        tty.die(
+            'Environment DAG is broken due to a missing package or repo.',
+            'To resolve without it, force concretize with the command: ',
+            '   spack --no-env-view concretize --force')
```

Comment:
```suggestion
            'Environment view is broken due to a missing package or repo.\n',
            '  To remove it and resolve the issue, ',
            'force concretize with the command:\n',
            '    spack concretize --force')
```

Reply:
We use DAG to refer to the dependency graph among packages, I think calling it the `Environment view` will be less confusing to users here.

Reply:
Ok, that makes sense. I think technically the DAG would be broken too though because the package it concretized with no longer exists.

Reply:
We can't recreate it, but as long as we don't need to look up package information (like recalculating the run environment) we can still use that DAG. For example, the user could use `spack env loads` to generate a script that loads all the modules, and the DAGs will still be valid when used in that way.